server/routers/work-shares/activateWallet.ts

```diff
@@ -33,18 +41,15 @@ export const activateWallet = protectedProcedure
     const workKeySharePromise = ctx.prisma.workKeyShare.findFirst({
       where: {
         userId: ctx.user.id,
-        sessionId: ctx.session.id,
+        deviceNonce: ctx.session.deviceNonce,
```

Comment:
`sessionId` was removed and `deviceNonce` was used because after signout `sessionId` is null and it couldn't find the workKeyShare for the wallet.

Reply:
Nice one! Also, before it was not properly using the index :\